tests/phpunit/elementor/modules/atomic-widgets/test-style-rendering.php

```diff
@@ -0,0 +1,248 @@
+									'unit' => 'px'
+								]
+							],
+							'text-decoration' => [
+								'$$type' => 'text-decoration',
+								'value' => 'underline'
+							]
+                        ],
+                        'meta' => [],
+                    ],
+                ],
+            ],
+        ];
+        $transformers = [
+            'size' => function($value) {
+				$size = (int)$value['size'];
+				$unit = $value['unit'];
+				return $size . $unit;
+            },
+        ];
+        $breakpoints = [];
+
+        $styleRender = new Style_Render( $styles, $transformers, $breakpoints);
+
+        // Act.
+        $css = $styleRender->render();
+
+        // Assert.
+        $this->assertStringContainsString('.test-style{font-size:14px;text-decoration:unset}', $css);
+    }
+
+	public function test_multiple_style_definitions_and_variants() {
+		// Arrange.
+		$styles = [
+			[
+				'id' => 'test-style-1',
+				'type' => 'class',
+				'variants' => [
+					[
+						'props' => [
+							'color' => 'red',
+						],
+						'meta' => [],
+					],
+					[
+						'props' => [
+							'color' => 'purple',
+						],
+						'meta' => [
+							'breakpoint' => 'mobile',
+						],
+					],
+					[
+						'props' => [
+							'font-weight' => 'bold',
+							'color' => 'yellow',
+							'font-size' => [
+								'$$type' => 'size',
+								'value' => [
+									'size' => 14,
+									'unit' => 'px'
+								]
+							],
+						],
+						'meta' => [
+							'breakpoint' => 'tablet',
+						],
+					],
+				],
+			],
+			[
+				'id' => 'test-style-2',
+				'type' => 'class',
+				'variants' => [
+					[
+						'props' => [
+							'color' => 'blue',
+						],
+						'meta' => [],
+					],
+					[
+						'props' => [
+							'color' => 'green',
+							'font-size' => [
+								'$$type' => 'size',
+								'value' => [
+									'size' => 12,
+									'unit' => 'px'
+								]
+							],
+						],
+						'meta' => [
+							'breakpoint' => 'mobile',
+						],
+					],
+				],
+			],
+		];
+		$transformers = [
+			'size' => function($value) {
+				$size = (int)$value['size'];
+				$unit = $value['unit'];
+				return $size . $unit;
+			}
+		];
+		$breakpoints = [
+			'mobile' => [
+				'type' => 'max-width',
+				'width' => 768,
+			],
+			'tablet' => [
+				'type' => 'max-width',
+				'width' => 1024,
+			],
+		];
+
+		$styleRender = new Style_Render( $styles, $transformers, $breakpoints);
+
+		// Act.
+		$css = $styleRender->render();
+
+		// Assert.
+		$this->assertStringContainsString('.test-style-1{color:red}', $css);
+		$this->assertStringContainsString('@media(max-width:768px){.test-style-1{color:purple}}', $css);
+		$this->assertStringContainsString('@media(max-width:1024px){.test-style-1{font-weight:bold;color:yellow;font-size:14px}}', $css);
+		$this->assertStringContainsString('.test-style-2{color:blue}', $css);
+		$this->assertStringContainsString('@media(max-width:768px){.test-style-2{color:green;font-size:12px}}', $css);
```

Comment:
Maybe we can add snapshot testing:

https://github.com/spatie/phpunit-snapshot-assertions

Reply:
Done (I think, worked well locally will now see how it breaks everything on the CI 😎)